src/rpc/introduction.md

```diff
@@ -0,0 +1,39 @@
+# Remote Procedure Calls (RPC)
+
+Communication between services occurs over remote procedure calls. RPCs happen all the time in distributed systesm. You browser had to make at least one RPC to this website to obtain this webpage.
+
+TiKV, as a distributed system involving a number of nodes, uses RPCs to communicate between each node, as well as the placement driver and clients.
+
+As it turns out, exposing functionality as a remote interface isn't a trivial. Networks are unreliable, systems are diverse and evolving, a huge variety of languages and formats exist, and even things like encoding are hard!

```

Comment:
```suggestion
As it turns out, exposing functionality as a remote interface isn't trivial. Networks are unreliable, systems are diverse and evolving, a huge variety of languages and formats exist, and even things like encoding are hard!
```

I would add something like: trying to make all these issues transparent (i.e., treating RPCs like local calls) has proven to be impossible, but making all the networking stuff explicit makes the communication code verbose and error-prone